skmorf/tree/_unsup_criterion.pyx

```diff
@@ -133,35 +160,181 @@ cdef class TwoMeans(Criterion):
             The new ending position for which to move samples from the right
             child to the left child.
         """
-        # need to implement
-        pass
+        cdef SIZE_t pos = self.pos
+        cdef SIZE_t end = self.end
+
+        cdef const SIZE_t[:] sample_indices = self.sample_indices
+        cdef const DOUBLE_t[:] sample_weight = self.sample_weight
+
+        cdef SIZE_t i
+        cdef SIZE_t p
+        cdef SIZE_t k
+        cdef SIZE_t c
+        cdef DOUBLE_t w = 1.0
+
+        # Update statistics up to new_pos
+        #
+        # Given that
+        #   sum_left[x] +  sum_right[x] = sum_total[x]
+        # and that sum_total is known, we are going to update
+        # sum_left from the direction that require the least amount
+        # of computations, i.e. from pos to new_pos or from end to new_po.
+        if (new_pos - pos) <= (end - new_pos):
+            for p in range(pos, new_pos):
+                i = sample_indices[p]
+
+                if sample_weight is not None:
+                    w = sample_weight[i]
+
+                for k in range(self.n_outputs):
+                    self.sum_left[k, <SIZE_t> self.y[i, k]] += w
+
+                self.weighted_n_left += w
+
+        else:
+            self.reverse_reset()
+
+            for p in range(end - 1, new_pos - 1, -1):
+                i = sample_indices[p]
+
+                if sample_weight is not None:
+                    w = sample_weight[i]
+
+                for k in range(self.n_outputs):
+                    self.sum_left[k, <SIZE_t> self.y[i, k]] -= w
+
+                self.weighted_n_left -= w
+
+        # Update right part statistics
+        self.weighted_n_right = self.weighted_n_node_samples - self.weighted_n_left
+        for k in range(self.n_outputs):
+            for c in range(self.n_classes[k]):
+                self.sum_right[k, c] = self.sum_total[k, c] - self.sum_left[k, c]
```

Comment:
Same comment: https://github.com/neurodata/scikit-tree/pull/7/files#r1064891519

Reply:
ah! now that I'm reading your comments, this could/should be 1D